executor/analyze_test.go

```diff
@@ -1159,3 +1159,370 @@ func (s *testSuite10) TestAdjustSampleRateNote(c *C) {
+	tbl := h.GetTableStats(tableInfo)
+	col := tbl.Columns[1]
+	c.Assert(len(col.TopN.TopN), Equals, 2)
+	c.Assert(len(col.Buckets), Equals, 2)
+
+	// auto-analyze uses the table-level options
+	tk.MustExec("insert into t values (20)")
+	c.Assert(h.DumpStatsDeltaToKV(handle.DumpAll), IsNil)
+	c.Assert(h.Update(is), IsNil)
+	h.HandleAutoAnalyze(is)
+	tbl = h.GetTableStats(tableInfo)
+	col = tbl.Columns[1]
+	c.Assert(len(col.TopN.TopN), Equals, 3)
+	c.Assert(len(col.Buckets), Equals, 5)
+
+	// options changed after alter table
+	tk.MustExec("alter table t STATS_OPTIONS='{\"TOPN\":\"4\",\"BUCKETS\":\"4\"}'")
+	is = tk.Se.(sessionctx.Context).GetInfoSchema().(infoschema.InfoSchema) // refresh infoschema
+	table, err = is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
+	tableInfo = table.Meta()
+	c.Assert(tableInfo.StatsOptions, NotNil)
+	c.Assert(tableInfo.StatsOptions.TopN, Equals, uint64(4))
+	tk.MustExec("insert into t values (21)")
+	c.Assert(h.DumpStatsDeltaToKV(handle.DumpAll), IsNil)
+	c.Assert(h.Update(is), IsNil)
+	h.HandleAutoAnalyze(is)
+	tbl = h.GetTableStats(tableInfo)
+	col = tbl.Columns[1]
+	c.Assert(len(col.TopN.TopN), Equals, 4)
+	c.Assert(len(col.Buckets), Equals, 4)
+
+	// manual-analyze to combine table-level and sql-level options
+	tk.MustExec("analyze table t with 2 buckets")
+	tbl = h.GetTableStats(tableInfo)
+	col = tbl.Columns[1]
+	c.Assert(len(col.TopN.TopN), Equals, 4)
+	c.Assert(len(col.Buckets), Equals, 2) // buckets num is overwritten by sql-level options
+
+	// manual-analyze for multiple tables without options should use table-level options for each
+	tk.MustExec("drop table if exists t2")
+	tk.MustExec("create table t2(b int) STATS_TOPN=12")
+	for i := 0; i < 20; i++ {
+		tk.MustExec(fmt.Sprintf("insert into t2 values (%d)", i))
+	}
+	tk.MustExec("analyze table t,t2")
+	tbl = h.GetTableStats(tableInfo)
+	col = tbl.Columns[1]
+	c.Assert(len(col.TopN.TopN), Equals, 4)
+	is = tk.Se.(sessionctx.Context).GetInfoSchema().(infoschema.InfoSchema) // refresh infoschema
+	table2, err2 := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t2"))
+	c.Assert(err2, IsNil)
+	tableInfo2 := table2.Meta()
+	tbl2 := h.GetTableStats(tableInfo2)
+	col2 := tbl2.Columns[1]
+	c.Assert(len(col2.TopN.TopN), Equals, 12)
+
+	// alter table with empty
+	tk.MustExec("alter table t STATS_OPTIONS=''")
+	is = tk.Se.(sessionctx.Context).GetInfoSchema().(infoschema.InfoSchema) // refresh infoschema
+	table, err = is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
+	tableInfo = table.Meta()
+	c.Assert(tableInfo.StatsOptions, NotNil)
+	c.Assert(tableInfo.StatsOptions.TopN, Equals, uint64(4))
+
+	// alter table with default
+	tk.MustExec("alter table t STATS_OPTIONS=default")
+	is = tk.Se.(sessionctx.Context).GetInfoSchema().(infoschema.InfoSchema) // refresh infoschema
+	table, err = is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
```

Comment:
So `STATS_OPTIONS=''`  don't modifying any stats option while `STATS_OPTIONS=default` clears all the stats option? Maybe we can address the difference in doc or somewhere because the two look similar.